src/ceph.in

```diff
@@ -359,36 +359,56 @@ def wrap(s, width, indent):
 
     raise StopIteration
 
+
+def terminal_width():
+    if not sys.stdout.isatty():
+        return 80
+    try:
+        import shutil
+        sz = shutil.get_terminal_size((80, 20))
+        return sz.columns
+    except AttributeError:
+        import fcntl
+        import struct
+        import termios
+        arg = struct.pack('HHHH', 0, 0, 0, 0)
+        ret = fcntl.ioctl(sys.stdout.fileno(), termios.TIOCGWINSZ, arg)
+        _, w, _, _ = struct.unpack('HHHH', ret)
+        return w
+
+
 def format_help(cmddict, partial=None):
     """
     Formats all the cmdsigs and helptexts from cmddict into a sorted-by-
     cmdsig 2-column display, with each column wrapped and indented to
-    fit into 40 characters.
+    fit into (terminal_width / 2) characters.
     """
 
     fullusage = ''
     for cmd in sorted(cmddict.values(), key=descsort_key):
 
         if not cmd['help']:
             continue
-        flags = cmd.get('flags')
-        if (flags is not None and
-            (flags & (FLAG_OBSOLETE | FLAG_DEPRECATED)) != 0):
+        flags = cmd.get('flags', 0)
+        if flags & (FLAG_OBSOLETE | FLAG_DEPRECATED):
             continue
         concise = concise_sig(cmd['sig'])
         if partial and not concise.startswith(partial):
             continue
-        siglines = [l for l in wrap(concise, 40, 1)]
-        helplines = [l for l in wrap(cmd['help'], 39, 1)]
+        width = terminal_width() - 1  # 1 for the line between sig and help
+        sig_width = int(width / 2)
+        help_width = int(width / 2) + (width % 2)
+        siglines = [l for l in wrap(concise, sig_width, 1)]
+        helplines = [l for l in wrap(cmd['help'], help_width, 1)]
 
         # make lists the same length
         maxlen = max(len(siglines), len(helplines))
         siglines.extend([''] * (maxlen - len(siglines)))
         helplines.extend([''] * (maxlen - len(helplines)))
 
         # so we can zip them for output
-        for (s, h) in zip(siglines, helplines):
-            fullusage += '{0:40s} {1}\n'.format(s, h)
+        for s, h in zip(siglines, helplines):
+            fullusage += '{s:{w}s} {h}\n'.format(s=s, h=h, w=sig_width)
```

Comment:
awesome, I did *not* know python's string formatting accepted variable width.  That's great to know.